### ISSUE TITLE
Plugin depending player ignoring support

### ISSUE THREAD

**Issue by Phoenix616** [Phoenix616/BetterBeds#4]
Don't count AFK players. For example by hooking into Essentials or maybe even running our own AFK checks (although I don't really want to get into this, might be only something for future updates if it really is needed)

**Comment by MattSturgeon** [comment]
FYI I'm working on a [simple plugin](https://github.com/MattSturgeon/AFKBeds) that sets the `betterbeds.ignore` permission to `true` while [WhosAFK](https://github.com/DrOverbuild/WhosAFK) set's a player afk

**Comment by Phoenix616** [comment]
You could also just fork and implement it directly into the plugin ;)

**Comment by MattSturgeon** [comment]
> You could also just fork and implement it directly into the plugin ;)

Yes, however if you look at my approach you'll see it makes more sense as a seperate plugin for a few reasons; most importantly, the code is specific to two plugins, you may decide there are better alternatives to WhosAFK, or someone using WhosAFK may wish to use an alternative bed plugin.

Also this is the first time I've used Java or the Bukkit API, so it made more sense for me to work on something small and independant.

_EDIT: Thought about the WhosAFK from a different approach, and I have come up with a simpler method than my integration plugin in PR #6_

**Comment by Phoenix616** [comment]
Some more thoughts about this issue:

It should generally support ignoring people due to multiple different parameters, the most obvious ones being the player being afk or the player being vanished. For that each supported plugin should use a different maven module and the different ignore types be configurable in the config.

AFK status as well as vanishing would ideally be handled by a ServiceProvider similarly to how Vault provides a unified interface, maybe something to integrate into [Tresor](https://github.com/Minebench/Tresor/)?

**Comment by seema84** [comment]
has it been implemented since 2016 in the meantime (for example essentialsx)?
